org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/javadoc/JavadocContentAccess2.java

```diff
@@ -0,0 +1,2415 @@
+			}
+			if (rep != null) {
+				if (nextToCopy < i) {
+					buf.append(text.substring(nextToCopy, i));
+				}
+				buf.append(rep);
+				nextToCopy = i + 1;
+			}
+		}
+		if (nextToCopy < length) {
+			buf.append(text.substring(nextToCopy));
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void handleInlineTagElement(TagElement node) {
+		String name = node.getTagName();
+
+		if (TagElement.TAG_VALUE.equals(name) && handleValueTag(node)) {
+			return;
+		}
+
+		boolean isLink = TagElement.TAG_LINK.equals(name);
+		boolean isLinkplain = TagElement.TAG_LINKPLAIN.equals(name);
+		boolean isCode = TagElement.TAG_CODE.equals(name);
+		boolean isLiteral = TagElement.TAG_LITERAL.equals(name);
+
+		if (isLiteral || isCode) {
+			fLiteralContent++;
+		}
+		if (isCode) {
+			fBuf.append("<code>"); //$NON-NLS-1$
+		}
+
+		if (isLink || isLinkplain) {
+			handleLink(node.fragments());
+		} else if (isCode || isLiteral) {
+			handleContentElements(node.fragments(), true);
+		} else if (handleInheritDoc(node)) {
+			// handled
+		} else if (handleDocRoot(node)) {
+			// handled
+		} else {
+			//print uninterpreted source {@tagname ...} for unknown tags
+			int start = node.getStartPosition();
+			String text = fSource.substring(start, start + node.getLength());
+			fBuf.append(removeDocLineIntros(text));
+		}
+
+		if (isCode) {
+			fBuf.append("</code>"); //$NON-NLS-1$
+		}
+		if (isLiteral || isCode) {
+			fLiteralContent--;
+		}
+
+	}
+
+	private boolean handleValueTag(TagElement node) {
+
+		List<? extends ASTNode> fragments = node.fragments();
+		try {
+			if (!(fElement instanceof IMember)) {
+				return false;
+			}
+			if (fragments.isEmpty()) {
+				if (fElement instanceof IField && JdtFlags.isStatic((IField) fElement) && JdtFlags.isFinal((IField) fElement)) {
+					IField field = (IField) fElement;
+					return handleConstantValue(field, false);
+				}
+			} else if (fragments.size() == 1) {
+				Object first = fragments.get(0);
+				if (first instanceof MemberRef) {
+					MemberRef memberRef = (MemberRef) first;
+					if (memberRef.getQualifier() == null) {
+						SimpleName name = memberRef.getName();
+						IType type = fElement instanceof IType ? (IType) fElement : ((IMember) fElement).getDeclaringType();
+						while (type != null) {
+							IField field = type.getField(name.getIdentifier());
+							if (field != null && field.exists()) {
+								if (JdtFlags.isStatic(field) && JdtFlags.isFinal(field)) {
+									return handleConstantValue(field, true);
+								}
+								break;
+							}
+							type = type.getDeclaringType();
+						}
+					}
+				}
+			}
+		} catch (JavaModelException e) {
+
+		}
+
+		return false;
+	}
+
+	private boolean handleConstantValue(IField field, boolean link) throws JavaModelException {
+		String text = null;
+
+		ISourceRange nameRange = field.getNameRange();
+		if (SourceRange.isAvailable(nameRange)) {
+			CompilationUnit cuNode = SharedASTProvider.getInstance().getAST(field.getTypeRoot(), null);
```

Comment:
This doesn't compile against master. I'll need to change it to:
```java
CoreASTProvider.getInstance().getAST(field.getTypeRoot(), CoreASTProvider.WAIT_YES, new NullProgressMonitor());
```

So please rebase your branch (and squash your commits, and sign the remaining one)